test/gtest/graphapi_capi_mha_backward.cpp

```diff
@@ -0,0 +1,430 @@
+        ///////////////////////////////////////////////////////////////////////////
+        auto mm2 = MakeGapiVirtualTensorDesc(m_testN, m_testH, m_testS, m_testS);
+        MakeMatmul(m_realTensorMap[miopenTensorMhaDO]->m_gapiDesc,
+                   m_realTensorMap[miopenTensorMhaV]->m_gapiDesc,
+                   mm2);
+
+        DescriptorWrapperPtr pwS7, pwS8, pwS9, pwS10;
+        MakePointwise(
+            MIOPEN_POINTWISE_MUL, mm2, m_realTensorMap[miopenTensorMhaDescaleDO]->m_gapiDesc, pwS7);
+        MakePointwise(
+            MIOPEN_POINTWISE_MUL, pwS7, m_realTensorMap[miopenTensorMhaDescaleV]->m_gapiDesc, pwS8);
+        MakePointwise(MIOPEN_POINTWISE_MUL, pwS8, rnd, pwS9);
+
+        MakePointwise(MIOPEN_POINTWISE_MUL,
+                      pwS9,
+                      m_realTensorMap[miopenTensorMhaDropoutProbability]->m_gapiDesc,
+                      pwS10);
+
+        ////////////////
+        DescriptorWrapperPtr pwS11, pwS12, pwS13, mult2;
+        MakePointwise(MIOPEN_POINTWISE_MUL,
+                      m_realTensorMap[miopenTensorMhaDO]->m_gapiDesc,
+                      m_realTensorMap[miopenTensorMhaDescaleDO]->m_gapiDesc,
+                      pwS11);
+        MakePointwise(MIOPEN_POINTWISE_MUL,
+                      m_realTensorMap[miopenTensorMhaO]->m_gapiDesc,
+                      m_realTensorMap[miopenTensorMhaDescaleO]->m_gapiDesc,
+                      pwS12);
+
+        MakePointwise(MIOPEN_POINTWISE_MUL, pwS11, pwS12, mult2);
+
+        MakePointwise(MIOPEN_POINTWISE_MUL,
+                      mult2,
+                      m_realTensorMap[miopenTensorMhaDropoutProbability]->m_gapiDesc,
+                      pwS13);
+
+        auto sum0 = MakeGapiVirtualTensorDesc(m_testN, m_testH, m_testS, 1);
+        MakeReduction(MIOPEN_REDUCE_TENSOR_ADD, pwS13, sum0);
+
+        ////////////////// Center Part ////////////////////////////////////////////
+        ///////////////////////////////////////////////////////////////////////////
+        DescriptorWrapperPtr sub1;
+        MakePointwise(MIOPEN_POINTWISE_SUB, pwS10, sum0, sub1);
+
+        DescriptorWrapperPtr pwS14, pwS15, mult3;
+        MakePointwise(MIOPEN_POINTWISE_IDENTITY, sub1, nullptr, pwS14, false, m_attentionScale);
+        MakePointwise(MIOPEN_POINTWISE_MUL, pwS14, pwS3 /*output from left column*/, mult3);
+        MakeReduction(
+            MIOPEN_REDUCE_TENSOR_MAX, mult3, m_realTensorMap[miopenTensorMhaAmaxDS]->m_gapiDesc);
+        MakePointwise(MIOPEN_POINTWISE_MUL,
+                      mult3,
+                      m_realTensorMap[miopenTensorMhaScaleDS]->m_gapiDesc,
+                      pwS15);
+
+        ////////////////// Center-bottom, Right-bottom ////////////////////////////
+        ///////////////////////////////////////////////////////////////////////////
+        auto mm3 = MakeGapiVirtualTensorDesc(m_testN, m_testH, m_testS, m_testD);
+        MakeMatmul(pwS15, m_realTensorMap[miopenTensorMhaK]->m_gapiDesc, mm3);
```

Comment:
@Vsevolod1983 The first `matmul` with input `K` is above and was using `K` as if it is already transposed. This is the second one which takes not transposed `K` and if `K` is already transposed we should add a `reshape` node before `matmul` either here to untranspose `K` or before the first `matmul` to flip the assumption about transposing of `K` and to stick closer to the graph from the docs.

Reply:
Once we get reshape node merged I will add it. Probably worth to discuss everything though